### PR TITLE
added npm deploy

### PR DESCRIPTION
**Pull request by andrewjtait**
Auto deploys to npm when a tag of `v#.#.#` is pushed.

Tested a tag of `v0.0.100` ✅ 
Tested a tag of `v0.0.100-rc1` ✅ 
Tested a normal tag ✅ 
Tested a branch ✅ 
Tested `release` branch ✅

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,10 +2,10 @@ language: node_js
 
 cache:
   directories:
-  - node_modules
+    - node_modules
 
 node_js:
-  - '8.9.2'
+  - 8.9.2
```
Comment (ianoxley) [inline .travis.yml]: The current LTS version is 8.9.4. Should we update this to match?

File: .travis.yml [hunk 2]
```diff
@@ -15,16 +15,24 @@ install:
...
+    api_key:
+      secure: nJbICzvxOsnsCQ+vpeLPyBNxTtNwAvLx53eBDaidkYXJD2xE5BYAmmkhk25lPK+o4tmK55mwvzvYh6NjjKO167HXnIRpiMcLhrzN+93jbueGD8Y92g3zvYC++8lRO64Xed122X+jyyhN/e5+immWNH/opklW4FnNQ1VhUh/uGTIDGyifz1Xqo9ozcMeYZfBSUdjPWDQmkDB052wsWG3yixOHsXwRJ4fAM19gtuzmKuqg8CHl5KLOmnf81WSkTp6W3yQpKXIvqJGjp4v/Zyoa0RISe0H+qlRhyY1BgZO4aZNaLrC23erBgss/Iw/EJKn+yFFIZUG8mMVCp9QgduCF0afPuxQ3g2fJCQX5+hRG5XjGMjHB2umbIgGfcA384exkdnlvXdmG5zLOh0LyEuqAI+5dm/6QvY6mMlOIETklS2m4iZ9x8ksw+o39XyYcSSIVpTSCZ1N3jIQ1sO+dV1QKr0Gx8t+nVmC5n5TqfLU+xxelp3dEvZFi5yZE3pcTE96T5ixL+Ad3nx4Hc2APv6OTuU4F/8gxDdbPnQP8Xj84QHOmBEEjoS4bsin74PvLW5atkjZSUr25j/VDWggnfwAxEN3RVb88iYH5ZfqoXEDcqhg0sX3ihUCRhA1pEFozoum4k0d09zG/hRp+h4m9xh9MfcgJIW/725+rOUrOB9z6NyU=
```
Comment (ianoxley) [inline .travis.yml]: Should these be in a publicly accessible repository?

File: .travis.yml (hunk 1 above)
Comment (andrewjtait) [reply .travis.yml]: Probably needs npm etc upgrading too. Would rather update and test that as a separate ticket.

Comment (andrewjtait) [reply .travis.yml]: ..done it anyway - guess the build will tell us what we need to know!

**Pull request merged**